### PR TITLE
[AIRFLOW-1840] Support back-compat on old celery config

### PR DESCRIPTION
**Pull request by ashb**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1840

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
    The new names are in-line with Celery 4, but if anyone upgrades Airflow without following the UPDATING.md instructions (which we probably assume most people won't, not until something stops working) their workers would suddenly just start failing. That's bad.

  This will issue a warning but carry on working as expected. We can remove the deprecation settings (but leave the code in config) after this release has been made.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
  new tests added to tests.configuration

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by ashb** [comment]
@bolkedebruin PTAL - I think we should have this in 1.10.

File: airflow/configuration.py [hunk 1]
```diff
@@ -121,9 +121,29 @@ class AirflowConfigParser(ConfigParser):
...
-        ('celery', 'result_backend')
+        ('celery', 'result_backend'),
+        # Todo: remove this in Airflow 1.11
+        ('celery', 'celery_result_backend'),
     }
 
+    # A two-level mapping of (section -> new_name -> old_name). When reading
+    # new_name, the old_name will be checked to see if it exists. If it does a
+    # DeprecationWarning will be issued and the old name will be used instead
+    deprecated_options = {
```
Comment (afernandez) [inline airflow/configuration.py:132]: How are multi-hop upgrades handled?
Say Airflow 1.8 -> 1.11

**Review by afernandez — APPROVED** [review]
+1 LGTM

File: airflow/configuration.py (hunk 1 above)
Comment (artwr) [reply airflow/configuration.py:132]: @afernandez Traditionally, we have not supported them very well. We have a few pre apache releases that required pretty much a full reinstall. Right now, I think we should mostly offer to support one hop upgrades.

Comment (ashb) [reply airflow/configuration.py:132]: Yeah - right now one hop is the most we should aim for. One of my long-term tasks is to write down what the back-compat policy we should adopt is. (I plan on basing it on Django's when I get around to it)

**Comment by ashb** [comment]
Retesting - failed half the jobs due to timeout issues.

**Comment by ashb** [comment]
Oh no, `assertWarns` is not available in python 2.7 :(

**Comment by ashb** [comment]
Transient Travis faliures now :/ Retesting

**Comment by artwr** [comment]
@ashb, any chance you could just test on your own fork and just post the link to the build success? We have been asked by infra to reduce our usage of the Apache Travis resources...

**Comment by ashb** [comment]
Ah okay, sorry. The last two of these were "it works, just one of the jobs timed out" - should I do that on my fork too?

(I thought Travis was free for open source projects...?)

**Comment by artwr** [comment]
It is for individual forks. I believe that the Apache foundation has to pay given the volume of builds.

File: airflow/configuration.py (hunk 1 above)
Comment (bolkedebruin) [reply airflow/configuration.py:132]: Breaking changes should happen only in major versions. Thus not from 1.10 -> 1.11, but rather from 1.10 -> 2.0

**Comment by mascah** [comment]
This looks like it's missing the option for `celery_result_backend` -> `result_backend`

**Comment by ashb** [comment]
Yup :(

**Pull request closed without merging**